### ISSUE TITLE
h5py: reading issue for groupBased file

### ISSUE THREAD

**Issue by ax3l** [openPMD/openPMD-viewer#299]
With `h5py`, both as backend in the current `dev` branch as well as using it through the last stable release, I have problems opening the following file through the slider: 
[openpmd_gzip.h5.gz](https://github.com/openPMD/openPMD-viewer/files/5718408/openpmd_gzip.h5.gz)

The openPMD-validator check does not raise errors on the file.

Since the problem does not appear with the `dev` branch using the `openPMD-api` backend, I think there might be a logic error in the `h5py` backend we need to address.

Errors in this notebook:
[openPMD-visualization.ipynb.gz](https://github.com/openPMD/openPMD-viewer/files/5718423/openPMD-visualization.ipynb.gz)

cc @RemiLehe 

## Debugging Progress

GUI: The failing `find_dataset` call tries to interpret scalar records as vector records, e.g.:
```
find_dataset: <HDF5 file "openpmd.h5" (mode r)> - Bx/x
find_dataset: <HDF5 file "openpmd.h5" (mode r)> - EypBx/x
```

While things like `Bx`, `By`, ... will become vector records eventually - others like `EypBx` stay scalar records.

API call: Running
```py
ts.get_field("EypBx", coord=None, iteration=9, plot=True)
```
gets the `find_dataset` right
```
find_dataset: <HDF5 file "openpmd.h5" (mode r)> - EypBx
```
but reads the zero (0) values from `iteration=0` even if one requests `iteration=9`.
